internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -191,3 +209,64 @@ func (c *CodePipeline) getPipelineName(resourceArn string) (string, error) {
 
 	return parsedArn.Resource, nil
 }
+
+// GetPipelineStatus retrieves status information from a given pipeline.
+func (c *CodePipeline) GetPipelineState(name string) (*PipelineState, error) {
+	input := &cp.GetPipelineStateInput{
+		Name: aws.String(name),
+	}
+	resp, err := c.client.GetPipelineState(input)
+
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline state %s: %w", name, err)
+	}
+	var stageStates []*StageState
+	for _, stage := range resp.StageStates {
+		var stageName string
+		if stage.StageName != nil {
+			stageName = aws.StringValue(stage.StageName)
+		}
+		var transition string
+		if stage.InboundTransitionState != nil {
+			transition = "DISABLED"
+			if *stage.InboundTransitionState.Enabled {
+				transition = "ENABLED"
+			}
+		}
+		var status []StageAction
```

Comment:
nit:
```suggestion
		var actions []StageAction
```

Reply:
done in https://github.com/aws/amazon-ecs-cli-v2/pull/1051